src/pytest_inmanta_lsm/load_generator.py

```diff
@@ -0,0 +1,203 @@
+    :contact: code@inmanta.com
+    :license: Inmanta EULA
+"""
+
+import asyncio
+import datetime
+import functools
+import logging
+import threading
+import time
+import typing
+
+from inmanta.data import model
+
+from pytest_inmanta_lsm import remote_orchestrator
+
+LOGGER = logging.getLogger(__name__)
+
+
+class LoadException(Exception):  # noqa: N818
+    pass
+
+
+class LoadGenerator:
+    def __init__(
+        self,
+        remote_orchestrator: remote_orchestrator.RemoteOrchestrator,
+        service_type: str,
+        logger: logging.Logger = LOGGER,
+        sleep_time: float = 1.0,
+    ):
+        self.remote_orchestrator = remote_orchestrator
+        self.service_type = service_type
+        self.sleep_time = sleep_time
+        self.logger = logger
+        self.running = True
+        self._thread: typing.Optional[threading.Thread] = None
+        self.exception: typing.Optional[Exception] = None
+
+    def __enter__(self) -> None:
+        self.logger.debug("Creating new Thread")
+        self._thread = threading.Thread(target=self.between_callback, daemon=True, name="Thread-LG")
+        self.logger.debug("Starting %s", self._thread.name)
+        self._thread.start()
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self.stop()
+        self.logger.debug("Stopping %s", self._thread.name)
+        self._thread.join(self.remote_orchestrator.client.timeout)
+        self.logger.debug("%s has been stopped", self._thread.name)
+
+        if self.exception is not None:
+            raise self.exception
+
+    @property
+    def thread(self) -> threading.Thread:
+        if self._thread is None:
+            raise RuntimeError("The thread is not initialized, this instance should be used in a context!")
+
+        return self._thread
+
+    def is_running(self) -> bool:
+        """
+        Retrieve the status of whether the thread should keep running or not
+        """
+        return self.running
+
+    def stop(self) -> None:
+        """
+        Set the `running` flag to False
+        """
+        self.running = False
+        self.logger.debug("%s should stop", self.thread.name)
+
+    def between_callback(self) -> None:
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        try:
+            loop.run_until_complete(self.create_load())
+        except LoadException:
+            self.logger.warning("%s - Load has been stopped!", self.thread.name)
+        except Exception as e:
+            self.exception = e
+        loop.close()
+
+    async def remote_call(self, call: typing.Callable[[], typing.Awaitable[model.BaseModel]]) -> None:
+        try:
+            await call()
+        except Exception as e:
+            self.logger.warning("%s - encountered the following error:%s!", self.thread.name, str(e))
+        finally:
+            if not self.is_running():
+                raise LoadException()
+
+            time.sleep(self.sleep_time)
```

Comment:
For this method, you should rather use `asyncio.sleep`, as the code base is async (even though it is true that blocking the thread is not a big issue, it is cleaner).
Also, it would be good to check `is_running` before calling the api (or after the sleep), and maybe only sleep for small intervals (i.e. half a second) at a time, always checking for `is_running` at each iteration.